stable/traefik/templates/configmap.yaml

```diff
@@ -100,18 +105,21 @@ data:
       {{- if .Values.kubernetes.ingressClass }}
     ingressClass = {{ .Values.kubernetes.ingressClass | quote }}
       {{- end}}
-      {{- if .Values.kubernetes.ingressEndpoint }}
-      [kubernetes.ingressEndpoint]
-        {{- if .Values.kubernetes.ingressEndpoint.hostname }}
-      hostname = {{ .Values.kubernetes.ingressEndpoint.hostname | quote }}
-        {{- end}}
-        {{- if .Values.kubernetes.ingressEndpoint.ip }}
-      ip = {{ .Values.kubernetes.ingressEndpoint.ip | quote }}
-        {{- end}}
-        {{- if .Values.kubernetes.ingressEndpoint.publishedService }}
-      publishedService = {{ .Values.kubernetes.ingressEndpoint.publishedService | quote }}
-        {{- end}}
+    {{- end}}
+    {{- if .Values.kubernetes.ingressEndpoint }}
+    [kubernetes.ingressEndpoint]
+      {{- if .Values.kubernetes.ingressEndpoint.hostname }}
+    hostname = {{ .Values.kubernetes.ingressEndpoint.hostname | quote }}
       {{- end}}
+      {{- if .Values.kubernetes.ingressEndpoint.ip }}
+    ip = {{ .Values.kubernetes.ingressEndpoint.ip | quote }}
+      {{- end}}
+      {{- if .Values.kubernetes.ingressEndpoint.publishedService }}
+    publishedService = {{ .Values.kubernetes.ingressEndpoint.publishedService | quote }}
+      {{- end}}
+    {{- else }}
```

Comment:
There should be no else though. If not defined, don't create an ingressEndpoint section.